internal/config/storage.go

```diff
@@ -59,6 +59,21 @@ type Local struct {
 
 // Git contains configuration for referencing a git repository.
 type Git struct {
-	Repository string `json:"repository,omitempty" mapstructure:"repository"`
-	Ref        string `json:"ref,omitempty" mapstructure:"ref"`
+	Repository string    `json:"repository,omitempty" mapstructure:"repository"`
+	Ref        string    `json:"ref,omitempty" mapstructure:"ref"`
+	BasicAuth  BasicAuth `json:"basicAuth,omitempty" mapstructure:"basic_auth"`
+	TokenAuth  TokenAuth `json:"tokenAuth,omitempty" mapstructure:"token_auth"`
```

Comment:
would it be possible to make this generic in someway? maybe similar to how we did it for the `AuthenticationConfig` ? https://github.com/flipt-io/flipt/blob/main/internal/config/authentication.go

Or some other way to specify that only one of these should be provided? Its not clear by looking at the struct (and expanding to the env vars/yaml) which version of the auth would take precedence here over the other

Reply:
@markphelps Yeah agreed generalizing this structure makes a ton of sense.